### PR TITLE
Refactored Routing

### LINKED ISSUE #162: Refactor Secure Routing
Right now, each component that requires the user to be signed in checks for authentication on their own. This isn't scalable, and we should check for authentication in the highest component possible

Tasks:
- Use array.map for rendering `<Route />`
- Use [PrivateRoute](https://github.com/evanugarte/SmartPhotoCategorizer/blob/master/src/PrivateRoute.js) to send in protected routes and check if authenticated
  - Example use [here](https://github.com/evanugarte/SmartPhotoCategorizer/blob/master/src/Routing.js)
- Get rid of authentication checks in all other components.

### PR DESCRIPTION
**Pull request by evanugarte**
We now check for the user's auth status in the highest component and use a
custom PrivateRoute component for securing routes. We also now use array.map in Routing.js to render the routes.

**Why are there so many other files changed?** When writing tests for the Router and PrivateRoute, the imported components had their own errors.

Resolves #162

### CODE REVIEW

**Comment by ThaiQ** [comment]
Why is there a white margin around the page? css is messed up?
image

CSS is messed up, Everything is centered:
image
before : 
image

another example 
image
before:
image

These are not show up when access "home" from admin dashboard
image
image

File: src/Pages/TheTeam/App.js [hunk 1]
```diff
@@ -151,62 +139,62 @@ export default class Example extends React.Component {
...
-            </Media>
-            <Media className='description'>
-              Cras sit amet nibh libero, in gravida nulla. Nulla vel metus
-              scelerisque ante
-            </Media>
-          </Card>
-        </Col>
-      )
-    }
+    // function block (target, index) {
```
Comment (ThaiQ) [inline src/Pages/TheTeam/App.js]: why are these commented out?

File: src/Components/Dashboard/Dashboard.js [hunk 2]
```diff
@@ -54,83 +21,81 @@ export class Dashboard extends Component {
...
+            </NavLink>
+            <NavLink
+              title='Overview'
+              to='/dashboard'
+              exact
+              activeClassName='active-nav-link'
+            >
+              Overview
+            </NavLink>
+            {this.state.user.accessLevel >= 2 && (
```
Comment (ThaiQ) [inline src/Components/Dashboard/Dashboard.js]: Admin does't show up anymore

File: src/Pages/Admin/Admin.js [hunk 3]
```diff
@@ -10,39 +9,8 @@ export class Admin extends Component {
...
-          this.props.history.push('/dashboard')
-        }
-      })
-      .catch(() => {
-        if (this.props.history) this.props.history.push('/login')
-      })
-  }
-
   render () {
-    return <div>{this.state.user.accessLevel >= 2 && <div>Admin</div>}</div>
```
Comment (ThaiQ) [inline src/Pages/Admin/Admin.js]: this is necessary to enable access to profile and admin in stead of the login button. See the last section of my comment.

**Comment by evanugarte** [comment]
Thanks for pointing it out. I will fix this and let you know when it is done

File: src/Pages/Admin/Admin.js (hunk 3 above)
Comment (evanugarte) [reply src/Pages/Admin/Admin.js]: I removed this as the `<PrivateRoute />` component takes care of this for you. In Routing.js, we see the Admin component get rendered on [this condition](https://github.com/SCE-Development/Core-v4/blob/aef4801af88c32c34bad2aba7ba09319db3e183a/src/Routing.js#L31). A [test](https://github.com/SCE-Development/Core-v4/blob/aef4801af88c32c34bad2aba7ba09319db3e183a/test/frontend/Routing.test.js#L217) also proves that with insufficient priviledges the user cannot see admin.

File: src/Components/Dashboard/Dashboard.js (hunk 2 above)
Comment (evanugarte) [reply src/Components/Dashboard/Dashboard.js]: Fixed.

File: src/Pages/3DPrintingConsole/app3DConsole.js [hunk 4]
```diff
@@ -30,32 +30,11 @@ export default class Example extends React.Component {
...
-      .catch(() => {
-        if (this.props.history) this.props.history.push('/login')
-      })
-
-    this.callDatabase()
+    this.setState({
+      isLoggedIn: true,
+      authToken: window.localStorage && window.localStorage.getItem('jwtToken')
+    })
+    if (window.localStorage) this.callDatabase()
```
[the remark this answers is not in the document]
Comment (evanugarte) [reply src/Pages/3DPrintingConsole/app3DConsole.js]: Done.

**Comment by ThaiQ** [comment]
Error in 3D-Printing Page
image
Error in console:
image

**Comment by ThaiQ** [comment]
I approved! All good but travis is failing bruh!

**Comment by evanugarte** [comment]
Everything should be good to go now. Further refactoring comes in #171 for authentication

**Comment by evanugarte** [comment]
Closing this. Will reopen the PR in a seperate branch,

**Pull request closed without merging**